### DISCUSSION TITLE
Do these models take in DICOM files?

### DISCUSSION THREAD

**Discussion by lesliewubbel** [Project-MONAI/MONAI#1593] [Q&A]
**Please use MONAI's Discussions tab**
For questions relating to MONAI usage, please do not create an issue.

Instead, use [MONAI's GitHub Discussions tab](https://github.com/Project-MONAI/MONAI/discussions). This can be found next to Issues and Pull Requests along the top of our repository.

Greetings Monai team,

I read awhile ago that Monai doesn't currently support DICOM files is that true? if so how could I help develop a pipeline line 
I am working on some code and small project to convert DICOM to .nii which c++ code already exist to support that but I mean more specifically for data pipeline for monai. I am really impressed by the project! thank you!

Leslie

**Comment by Nic-Ma** [comment]
Hi @Gresliebear ,

Thanks for your interest here.
MONAI can load DICOM images from v0.4, you can check this tutorial for more details:
https://github.com/Project-MONAI/tutorials/blob/master/modules/load_medical_images.ipynb

Thanks.

**Comment by oscarramirezs** [comment]
Hi Nic-Ma

thank you for the réponse, 

im trying to use dice files but the matrix en dicom are 512*512*deep and nifty are deep*512*512, how can I figure-out a solution for this issue?

Oscar

**Comment by lesliewubbel** [comment]
Thank you Nic!

**Comment by Nic-Ma** [comment]
> Hi Nic-Ma
> 
> thank you for the réponse,
> 
> im trying to use dice files but the matrix en dicom are 512_512_deep and nifty are deep_512_512, how can I figure-out a solution for this issue?
> 
> Oscar

Hi @ericspod ,

I am not very familiar with DICOM details, could you please help share some comments on this question?

Thanks in advance.

**Comment by lesliewubbel** [comment]
@Nic-Ma  

Hes talking about when a DICOM file or a stack of DICOM files created into lets say a python object we can pull its metadata for the shape 

DICOM_object.shape  
prints out (512,512, 200)  which means that each dicom file or 'image' or 'slice' has a height and width of 512 pixels. The 200 is the number of slices or how "deep" the whole stack. so logical we know (x,y,z) but nifty files orientation are flipped (z,x,y)
![image](https://user-images.githubusercontent.com/13157730/107867217-01640e80-6e47-11eb-997f-757fe3a3f4e9.png)

https://www.semanticscholar.org/paper/The-first-step-for-neuroimaging-data-analysis%3A-to-Li-Morgan/744629069861e2a6af9f793268ce8240181b54d1/figure/0
Nifty files flips to (200, 512, 512) what happened is that the spatial orientation changed because DICOM files are read by radiologist who used left-handed view in their spatial orientation. When neurologists read right hand in their spatial orientation therefore nitfy files are just the inverse of 3D DICOM files. The reason being nii files were created originally for neurologist looking at MRI or fMRI data.


![image](https://user-images.githubusercontent.com/13157730/107867229-16d93880-6e47-11eb-85d7-8373d5fac7f1.png)


to convert DICOM to nifty use the dicom2nitfi package https://github.com/icometrix/dicom2nifti

**Comment by rijobro** [comment]
There is also `np.moveaxis` and MONAI's `Flip` should you want to reorder dimensions and their directions, respectively.

**Comment by ericspod** [comment]
@Nic-Ma , @Gresliebear 's description is what I've found with Nifti coordinate spaces and the RAS/LAS orientations. DICOM files generally represent the image orientation as two vectors in a common scanner space. The rotation between Nifti-space and that common space is pi/2 around the Z axis as his figure shows.

**Comment by lesliewubbel** [comment]
@ericspod  thank you for the information! I am self-taught so the terminology can be confusing if something is wrong in my description please point it out.

**Comment by ericspod** [comment]
I've always just been informal about the DICOM orientations, X is to the left, Y is to the back, Z is to the head.

**Comment by Nic-Ma** [comment]
@ericspod , thanks for your information, so what's the next steps? Should we modify anything?

Thanks.

**Comment by aylward** [comment]
There is a nice explanation of DICOM vs "Laboratory/Display" orientation
and Neurosurgery vs Radiological image display at
https://itk.org/Wiki/Proposals:Orientation#Some_notes_on_the_DICOM_convention_and_current_ITK_usage



On Mon, Feb 15, 2021 at 8:09 PM Eric Kerfoot <notifications@github.com>
wrote:

> I've always just been informal about the DICOM orientations, X is to the
> left, Y is to the back, Z is to the head.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Project-MONAI/MONAI/issues/1580#issuecomment-779520389>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACEJL7SE7PWVKO232WMN3LS7HAS7ANCNFSM4XREKNBA>
> .
>


-- 
Stephen R. Aylward, Ph.D.
Senior Director of Strategic Initiatives
---
Kitware: *Delivering innovative, open source, scientific software.*

**Comment by oscarramirezs** [comment]
it is possible to add to the monai.transforms a fonction such => image= torch.transpose(image, 0, 1)??
